src/blockdev.rs

```diff
@@ -1515,3 +1516,270 @@ mod tests {
         );
     }
 }
+
+#[derive(Debug, Deserialize)]
+struct BlockDevices {
+    pub blockdevices: Vec<Device>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct Device {
+    pub name: String,
+    pub label: Option<String>,
+    pub uuid: Option<String>,
+    pub children: Option<Vec<Device>>,
+}
+
+fn count_partition_by_label(label: &str, devices: &[Device]) -> usize {
+    fn disks_flatten<'a>(root: &'a [Device], all: &mut HashMap<&'a str, HashSet<&'a str>>) {
+        for device in root {
+            if let Some(label) = device.label.as_ref() {
+                let uuid = match device.uuid.as_ref() {
+                    Some(v) => v.as_str(),
+                    _ => device.name.as_str(),
+                };
+                all.entry(label).or_default().insert(uuid);
+            }
+            if let Some(children) = device.children.as_ref() {
+                disks_flatten(children, all);
+            }
+        }
+    }
+
+    let mut all = HashMap::new();
+    disks_flatten(devices, &mut all);
+    eprintln!("{:?}", all);
```

Comment:
Hmm, could we keep all the printing in the higher-level functions and avoid printing anything in the utility functions? Maybe instead the API can be `get_partitions_with_label()` and it returns a `Vec`? And then the `install` function can print out the elements if necessary.

Reply:
ooops, forgot to remove. that was just for debugging purposes

>Maybe instead the API can be get_partitions_with_label() and it returns a Vec? 

ideally yes, i wanted to do that after #631